src/quickpicks/schemaRegistries.ts

```diff
@@ -1,45 +1,43 @@
 import * as vscode from "vscode";
 import { IconNames } from "../constants";
-import { getEnvironments } from "../graphql/environments";
 import { Logger } from "../logging";
 import { CCloudEnvironment } from "../models/environment";
 import { CCloudSchemaRegistry, SchemaRegistry } from "../models/schemaRegistry";
 import { hasCCloudAuthSession } from "../sidecar/connections";
+import { CCloudResourcePreloader } from "../storage/ccloudPreloader";
 import { getResourceManager } from "../storage/resourceManager";
+import { getSchemasViewProvider } from "../viewProviders/schemas";
 
 const logger = new Logger("quickpicks.schemaRegistry");
 
-/** Progress wrapper for the Schema Registry quickpick to accomodate data-fetching time. */
-export async function schemaRegistryQuickPick(): Promise<SchemaRegistry | undefined> {
-  return await vscode.window.withProgress(
-    {
-      location: { viewId: "confluent-schemas" },
-      title: "Loading Schema Registries...",
-    },
-    async () => {
-      return await generateSchemaRegistryQuickPick();
-    },
-  );
-}
-
 /**
  * Create a quickpick to let the user choose a Schema Registry (listed by CCloud environment
  * separators). Mainly used in the event a command was triggered through the command palette instead
  * of through the view->item->context menu.
+ *
+ * @returns The selected Schema Registry, or undefined if none was selected.
  */
-async function generateSchemaRegistryQuickPick(): Promise<SchemaRegistry | undefined> {
+export async function schemaRegistryQuickPick(): Promise<SchemaRegistry | undefined> {
   // TODO(shoup): update to support LocalSchemaRegistry
-  if (!hasCCloudAuthSession()) {
-    return undefined;
-  }
+
+  // schema registries are a coarse resource, so ensure they are loaded before proceeding
+  const preloader = CCloudResourcePreloader.getInstance();
+  await preloader.ensureCoarseResourcesLoaded();
+
   // list all Schema Registries for all CCloud environments for the given connection; to be
   // separated further by environment in the quickpick menu below
-  const envGroups = await getEnvironments();
-  const ccloudSchemaRegistries: CCloudSchemaRegistry[] = envGroups
-    .map((group) => group.schemaRegistry)
-    .filter((registry) => registry !== undefined) as CCloudSchemaRegistry[];
+  const resourceManager = getResourceManager();
+  const ccloudSchemaRegistries = Array.from(
+    (await resourceManager.getCCloudSchemaRegistries()).values(),
+  );
+
   if (ccloudSchemaRegistries.length === 0) {
-    vscode.window.showInformationMessage("No Schema Registries available.");
+    let message = "No Schema Registries available.";
+
+    if (!hasCCloudAuthSession()) {
+      message += " Perhaps log into to Confluent Cloud first?";
```

Comment:
Let's add a button to make sure the user can quickly log in to CCloud if this is the case -- one example from the Kafka cluster quickpick: https://github.com/confluentinc/vscode/blob/0725d2ef121d495b9771b0bb7cf1a6fe6aa22acf/src/quickpicks/kafkaClusters.ts#L62-L69